### PR TITLE
feat: sfx to xbi conversion

### PR DESCRIPTION
**Pull request by palozano**
- chore: add implementation for sfx2xbi
- chore: move fns around
- chore: make compileable
- feat: enable an extrinsic for executing SFX
- chore: add aliq and swap
- chore: update all sfx xbi instrucs

# Summary of changes

Allow the execution of SideEffects on XBI by creating
the XBI instructions from the SideEffect info.

Addresses issue #672 

## Acceptance Checklist

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue):
  - [ ] Have you **written tests**to protect against future occurrences of the bug?
- [X] New feature (non-breaking change which adds functionality)
  - [ ] Have you **seen it working** in a development environment?
  - [ ] Have you **written tests** for **happy** and **unhappy** paths?
  - [ ] Have you implemented this feature as per the **issue acceptance criteria**?
- [ ] Breaking change (a feature that would cause existing functionality not to work as expected
  - [ ] Is the breaking change **documented**?
  - [ ] Has any previous changes been **deprecated**?
- [ ] CI/CD
  - [ ] If introducing new actions, have you **looked at the action code** for anything spurious?
  - [ ] Have you written **custom scripts** for things that could be actions already on the marketplace?
  - [ ] If introducing new actions, have you **pegged a static version**?
- [ ] Documentation Update
  - [ ] Have you checked other documentation, such as the docs repository?
  - [ ] If updating script documentation, have you **tested it on both environments**?
- [ ] Substrate
  - [ ] RPC methods?
  - [ ] Chainspec, both JSON specs & the module?
  - [ ] Runtime versioning?
  - [ ] Benchmarks?
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by palozano** [comment]
Tests missing. Will write them.

File: pallets/executors/src/lib.rs [hunk 1]
```diff
@@ -1623,18 +1656,230 @@ pub mod pallet {
...
-            T::InstructionHandler::handle(origin, xbi)
+        pub fn execute_side_effects_with_xbi(
+            origin: OriginFor<T>, // Active relayer
+            xtx_id: XExecSignalId<T>,
+            side_effect: SideEffect<T::AccountId, BalanceOf<T>>,
+            max_exec_cost: u128,
+            max_notifications_cost: u128,
+        ) -> DispatchResultWithPostInfo {
+            let executor = ensure_signed(origin.clone())?;
+            let state = <CandidateInfo<T>>::get(&executor).ok_or(Error::<T>::NoSuchCandidate)?;
```
Comment (MaciejBaj) [inline pallets/executors/src/lib.rs:1667]: Don't think that check is necessary - executor as per latest design could be any valid account on t3rn.

File: pallets/executors/src/lib.rs [hunk 2]
```diff
@@ -1623,18 +1656,230 @@ pub mod pallet {
...
+            // validate extrinsic origin is actually an executor in the active set
+            ensure!(state.is_active(), Error::<T>::AlreadyActive);
+
+            let account_to_32: AccountId32 = Decode::decode(&mut &executor.encode()[..])
```
Comment (MaciejBaj) [inline pallets/executors/src/lib.rs:1671]: Rely on Sabi / Scabi from XBI to convert between T::AccountId to AccoutnId32

File: pallets/executors/src/lib.rs [hunk 3]
```diff
@@ -1623,18 +1656,230 @@ pub mod pallet {
...
+                .map_err(|_| Error::<T>::AlreadyActive)?;
```
Comment (MaciejBaj) [inline pallets/executors/src/lib.rs:1672]: Wrong Err

File: pallets/executors/src/lib.rs [hunk 4]
```diff
@@ -1623,18 +1656,230 @@ pub mod pallet {
...
+
+            let nonce_always_0_because_we_use_seed = 0;
+            let bypass_most_metadata_checks_default_para_id = Default::default();
+
+            let sfx_id = side_effect.generate_id::<T::Hashing>(xtx_id.as_ref(), 0u32); // FIXME: index needs to be passed from
+
+            let mut xbi: XbiFormat = SfxWithMetadataNewtype::<T>::new(
+                side_effect,
+                XbiMetadata::new(
+                    bypass_most_metadata_checks_default_para_id,
```
Comment (MaciejBaj) [inline pallets/executors/src/lib.rs:1682]: Take target para_id from sfx.target, source para_id from Circuit::self_para_id

File: pallets/executors/src/lib.rs [hunk 5]
```diff
@@ -1623,18 +1656,230 @@ pub mod pallet {
...
+    fn try_into(self) -> Result<XbiFormat, Self::Error> {
+        let side_effect = self.side_effect;
+        let metadata = self.metadata;
+
+        match &side_effect.action {
+            b"tran" => {
+                Ok(XbiFormat {
+                    instr: XbiInstruction::Transfer {
+                        // Get dest as argument_1 of SFX::Transfer of Type::DynamicAddress
+                        dest: Decode::decode(&mut &side_effect.encoded_args[0][..])
```
Comment (MaciejBaj) [inline pallets/executors/src/lib.rs:1733]: Again, try to rely on Sabi

File: pallets/executors/src/lib.rs [hunk 6]
```diff
@@ -1623,18 +1656,230 @@ pub mod pallet {
...
+                        ))
+                        .map_err(|_| Error::<T>::SfxDecodingValueErr)?,
+                    },
+                    metadata,
+                })
+            },
+            b"mult" | b"tass" => Ok(XbiFormat {
+                instr: XbiInstruction::TransferAssets {
+                    // Get dest as argument_0 of SFX::TransferAssets of Type::DynamicBytes
+                    currency_id: Decode::decode(&mut &side_effect.encoded_args[0][..])
```
Comment (MaciejBaj) [inline pallets/executors/src/lib.rs:1747]: Use Sabi

File: pallets/executors/src/lib.rs [hunk 7]
```diff
@@ -1623,18 +1656,230 @@ pub mod pallet {
...
+                        .map_err(|_| Error::<T>::SfxDecodingAddressErr)?,
+                    // Get dest as argument_1 of SFX::TransferAssets of Type::DynamicAddress
+                    dest: Decode::decode(&mut &side_effect.encoded_args[1][..])
```
Comment (MaciejBaj) [inline pallets/executors/src/lib.rs:1750]: Use Sabi

File: pallets/executors/src/lib.rs [hunk 8]
```diff
@@ -1623,18 +1656,230 @@ pub mod pallet {
...
+                        .map_err(|_| Error::<T>::SfxDecodingAddressErr)?,
+                    // Get dest as argument_2 of SFX::TransferAssets of Type::Value
+                    value: Sabi::try_convert(ValueMorphism::<_, u128>::new(
+                        &mut &side_effect.encoded_args[2][..],
+                    ))
+                    .map_err(|_| Error::<T>::SfxDecodingValueErr)?,
+                },
+                metadata,
+            }),
+            b"aliq" => Ok(XbiFormat {
```
Comment (MaciejBaj) [inline pallets/executors/src/lib.rs:1760]: Use Sabi

**Review by MaciejBaj — CHANGES_REQUESTED** [review]

File: pallets/executors/src/lib.rs (hunk 1 above)
Comment (palozano) [reply pallets/executors/src/lib.rs:1667]: Removed

**Comment by AwesomeIbex** [comment]
commit the lockfille

File: pallets/executors/src/lib.rs (hunk 2 above)
Comment (palozano) [reply pallets/executors/src/lib.rs:1668]: Isn't it bad to couple XBI to an executable dependency? This does the same and wouldn't conflict if there are breaking changes in XBI.

**Review by AwesomeIbex — COMMENTED** [review]
Lets merge this and then add unhappy path tests for all of those lines that codecov moans about

**Review by MaciejBaj — APPROVED** [review]

**Pull request merged**